zerver/lib/bot_lib.py

```diff
@@ -160,8 +161,26 @@ def send_reply(
             )
         return {"id": result["id"]}
 
-    def update_message(self, message: Dict[str, Any]) -> None:
-        pass  # Not implemented
+    def update_message(self, message: Dict[str, Any]) -> Dict[str, Any]:
+        if not self._rate_limit.is_legal():
+            self._rate_limit.show_error_and_exit()
+
+        # update_message receives "message_id" instead of "id" for the message to be updated.
+        # This can be confused with react which uses "id" as the key.
+        if "message_id" not in message:
+            raise EmbeddedBotKeyNotFound(_('Expected to find the "message_id" required key.'))
+
+        check_update_message(
+            user_profile=self.user_profile,
+            message_id=int(message["message_id"]),
+            stream_id=None,
+            topic_name=message.get("topic", None),
+            propagate_mode=message.get("propagate_mode", "change_one"),
+            send_notification_to_old_thread=message.get("send_notification_to_old_thread", False),
+            send_notification_to_new_thread=message.get("send_notification_to_new_thread", False),
+            content=message.get("content", None),
+        )
```

Comment:
I'd like us to do proper parsing here using `check_dict(...)` to make sure that we're not passing unexpected types for these parameters into `check_update_message`.